### PR TITLE
chore: identity checks for all

### PR DESCRIPTION
**Pull request by 35C4n0r**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes https://github.com/tailcallhq/tailcall/issues/1414
/claim #1414 

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [x] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **New Features**
    - Enhanced schema definitions with specific directives for optimized data retrieval and improved schema clarity.
    - Introduced new fields and modified existing ones in types like `Query`, `User`, and `Post` for better data fetching and relationship definition.
    - Optimized data fetching mechanisms and API behavior with the addition of `@upstream`, `@http`, and other directives.

- **Bug Fixes**
    - Corrected argument order in `@http` directive across types to address endpoint construction logic issues.
    - Streamlined schema declarations by eliminating redundancy and unnecessary nesting.

- **Refactor**
    - Reorganized schema declarations by adjusting field and type order to enhance overall clarity and structure.
    - Improved directive parameter adjustments and property reordering within schema definitions for enhanced configuration and readability.

- **Documentation**
    - Enhanced block syntax in `tests/README.md` to ensure clearer definition and consistency in test runner documentation.

- **Chores**
    - Conducted general cleanup in test files by removing extraneous content and consolidating schema annotations.

### CODE REVIEW

**Comment by 35C4n0r** [comment]
@tusharmath PR is ready for review.
PS: I've also updated the documentation for #1213

File: tests/README.md [hunk 1]
```diff
@@ -52,24 +52,23 @@ These are usually only added to tests temporarily, so try not to commit tests wi
...
 Example:
 
 ````md
-#### server:
 
-```graphql
+```graphql @server
 schema {
   query: Query
 }
```
[the remark this answers is not in the document]
Comment (tusharmath) [reply tests/README.md:74]: create a patch

File: tests/execution/add-field-index-list.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
 # Sending field index list
 
 ```graphql @server
-schema {
+schema @server @upstream {
```
Comment (tusharmath) [inline tests/execution/add-field-index-list.md:4]: We shouldn't need empty `@server` and `@upstream`.

File: tests/execution/apollo-tracing.md [hunk 3]
```diff
@@ -1,14 +1,12 @@
 # Apollo Tracing
 
 ```graphql @server
-schema
-  @server(port: 8000, graphiql: true, hostname: "0.0.0.0")
-  @telemetry(export: {apollo: {apiKey: "<api_key>", graphRef: "tailcall-demo-3@current"}}) {
```
Comment (tusharmath) [inline tests/execution/apollo-tracing.md:6]: Telemetry directive has been deleted.

File: tests/execution/add-field-index-list.md (hunk 2 above)
Comment (35C4n0r) [reply tests/execution/add-field-index-list.md:4]: Without this the identity checks will fail.

File: tests/execution/batching-default.md [hunk 4]
```diff
@@ -1,21 +1,20 @@
 # Batching default
 
 ```graphql @server
-schema @server @upstream(baseURL: "http://jsonplaceholder.typicode.com", httpCache: true, batch: {delay: 10}) {
+schema @server @upstream(baseURL: "http://jsonplaceholder.typicode.com", batch: {delay: 10, headers: [], maxSize: 100}, httpCache: true) {
```
Comment (tusharmath) [inline tests/execution/batching-default.md:4]: default values should not be serialized

File: tests/execution/batching-group-by-default.md [hunk 5]
```diff
@@ -1,23 +1,20 @@
 # Batching group by default
 
 ```graphql @server
-schema
-  @server
-  @upstream(baseURL: "http://jsonplaceholder.typicode.com", httpCache: true, batch: {delay: 1, maxSize: 1000}) {
+schema @server @upstream(baseURL: "http://jsonplaceholder.typicode.com", batch: {delay: 1, headers: [], maxSize: 1000}, httpCache: true) {
```
Comment (tusharmath) [inline tests/execution/batching-group-by-default.md:4]:
```suggestion
schema @server @upstream(baseURL: "http://jsonplaceholder.typicode.com", batch: {delay: 1, maxSize: 1000}, httpCache: true) {
```

Empty headers should not be inserted.

File: tests/execution/add-field-index-list.md (hunk 2 above)
Comment (tusharmath) [reply tests/execution/add-field-index-list.md:4]: Need to fix it in the config.rs

**Comment by tusharmath** [comment]
@35C4n0r Converting the PR to draft. Feel free to change it once the PR is ready with the fixes.

**Comment by tusharmath** [comment]
Is there work happening on this front? If not do you mind closing this PR?

**Pull request closed without merging**